api/nodes/serializers.py

```diff
@@ -23,88 +23,74 @@ class NodeSerializer(JSONAPISerializer):
     tags = ser.SerializerMethodField(help_text='A dictionary that contains two lists of tags: '
                                                'user and system. Any tag that a user will define in the UI will be '
                                                'a user tag')
+    registration = ser.BooleanField(read_only=True, source='is_registration')
+    collection = ser.BooleanField(read_only=True, source='is_folder')
+    dashboard = ser.BooleanField(read_only=True, source='is_dashboard')
 
-    links = LinksField({
-        'html': 'get_absolute_url',
+    links = LinksFieldWIthSelfLink({'html': 'get_absolute_url'})
+    # TODO: When we have 'admin' permissions, make this writable for admins
+    public = ser.BooleanField(source='is_public', read_only=True,
+                              help_text='Nodes that are made public will give read-only access '
+                                                            'to everyone. Private nodes require explicit read '
+                                                            'permission. Write and admin access are the same for '
+                                                            'public and private nodes. Administrators on a parent '
+                                                            'node have implicit read permissions for all child nodes',
+                              )
+
+    relationships = LinksField({
         'children': {
-            'related': Link('nodes:node-children', kwargs={'node_id': '<pk>'}),
-            'count': 'get_node_count',
+            'links': {
+                'related': {
+                    'href': Link('nodes:node-children', kwargs={'node_id': '<pk>'}),
+                    'meta': 'get_objects_data:children'
+                }
+            },
         },
         'contributors': {
-            'related': Link('nodes:node-contributors', kwargs={'node_id': '<pk>'}),
-            'count': 'get_contrib_count',
+            'links': {
+                'related': {
+                    'href': Link('nodes:node-contributors', kwargs={'node_id': '<pk>'}),
+                    'meta': 'get_objects_data:contributors'
+                }
+            },
         },
         'pointers': {
-            'related': Link('nodes:node-pointers', kwargs={'node_id': '<pk>'}),
-            'count': 'get_pointers_count',
+            'links': {
+                'related': {
+                    'href': Link('nodes:node-pointers', kwargs={'node_id': '<pk>'}),
+                    'meta': 'get_objects_data:pointers'
+                }
+            },
         },
         'registrations': {
-            'related': Link('nodes:node-registrations', kwargs={'node_id': '<pk>'}),
-            'count': 'get_registration_count',
+            'links': {
+                'related': {
```

Comment:
What would be super-cool is if you could just do `'related': Link('nodes:node-registrations', kwargs={'node_id': '<pk>'}),` here and have it return both the href and the meta (presuming there is meta information to return)